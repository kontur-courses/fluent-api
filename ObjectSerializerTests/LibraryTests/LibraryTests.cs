using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using ObjectSerializer;
using ObjectSerializerTests.ClassesToSerialize;

namespace ObjectSerializerTests.LibraryTests
{
    [TestFixture]
    public class LibraryTests
    {
        private Library serializeLibrary;

        [OneTimeSetUp]
        public void Init()
        {
            serializeLibrary = new Library("Имени кого-то там");

            serializeLibrary.AddBook(new Book("1984", "George Orwell"));
            serializeLibrary.AddBook(new Book("To Kill a Mockingbird", "Harper Lee"));
            serializeLibrary.AddBook(new Book("The Great Gatsby", "F. Scott Fitzgerald"));
            serializeLibrary.AddBook(new Book("Pride and Prejudice", "Jane Austen"));
        }

        [UseReporter(typeof(DiffReporter))]
        [Test]
        public void ObjectPrinter_CanPrint_CollectionObjects()
        {
            var printer = ObjectPrinter.For<Library>()
                .Exclude(l => l.BooksByAuthor);

            Approvals.Verify(printer.PrintToString(serializeLibrary));
        }

        [UseReporter(typeof(DiffReporter))]
        [Test]
        public void ObjectPrinter_CanPrint_DictionaryObjects()
        {
            var printer = ObjectPrinter.For<Library>()
                .Exclude(l => l.Books);

            Approvals.Verify(printer.PrintToString(serializeLibrary));
        }

        [UseReporter(typeof(DiffReporter))]
        [Test]
        public void ObjectPrinter_CanPrint_DictionaryAndCollectionCorrectly()
        {
            var printer = ObjectPrinter.For<Library>()
                .For(l => l.Name)
                .Using(n => $"Отличная библиотека {n}");

            Approvals.Verify(printer.PrintToString(serializeLibrary));
        }
    }
}
