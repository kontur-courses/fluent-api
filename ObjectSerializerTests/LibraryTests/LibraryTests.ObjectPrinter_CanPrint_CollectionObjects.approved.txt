BaseLibrary
	Books = 
		Collection
			Book
				Title = "1984"
				Author = "George Orwell"
			Book
				Title = "To Kill a Mockingbird"
				Author = "Harper Lee"
			Book
				Title = "The Great Gatsby"
				Author = "F. Scott Fitzgerald"
			Book
				Title = "Pride and Prejudice"
				Author = "Jane Austen"
	Name = "Имени кого-то там"